client/containers/Pub/PubDocument/PubMouseEvents/PubMouseEvents.tsx

```diff
@@ -1,95 +1,116 @@
-import React, { useEffect, useRef, useReducer } from 'react';
+import React, { useEffect, useReducer } from 'react';
 
-import HeaderPopover from './HeaderPopover';
+import { getHighestAncestorWithId } from 'client/utils/dom';
 
-type Props = {
-	collabData: any;
+import LinkPopover from './LinkPopover';
+
+export type PubMouseEventProps = {
 	locationData: any;
-	historyData: any;
 	mainContentRef: any;
 };
 
-/* Specify the types of elems we want events for */
-const mouseElemTypes = [
-	{
-		key: 'header',
-		querySelector: ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'].map(
-			(headerType) => `.pub-body-component > .editor >${headerType}`,
-		),
-	},
-];
-
-const PubMouseEvents = (props: Props) => {
-	const { collabData, historyData, mainContentRef, locationData } = props;
-	const timeouts = useRef({});
-	const [hoverElems, hoverElemsDispatch] = useReducer((state, action) => {
-		return {
-			...state,
-			[action.type]: action.elem,
+const clickToCopySelector = '.click-to-copy, .click-to-copy *';
+
+enum HoverTargetTypes {
+	ClickToCopy = 'click-to-copy',
+}
+
+type PubMouseEventState = {
+	[hoverTargetType: string]: Element | undefined;
+};
+
+type PubMouseEventAction = {
+	type: string;
+	element?: Element;
+};
+
+function isValidLinkTarget(element: Element | null): element is Element {
+	if (!element) {
+		return false;
+	}
+
+	if (element.tagName === 'P' && element.textContent === '') {
+		return false;
+	}
+
+	return true;
+}
+
+const PubMouseEvents = (props: PubMouseEventProps) => {
```

Comment:
Thoughts on renaming this component, now that it only handles the link copying functionality?

Reply:
`PubLinkController` was the best I could do. Any suggestions?